### ISSUE TITLE
Duplicate log lines created when redirecting STDOUT

### ISSUE THREAD

**Issue by NaomiReeves** [chef/chef#7184]
## Description
An additional stdout_logger is getting created when redirecting STDOUT.

It looks like it's only happening when STDOUT is not a tty and is likely related to the refactoring in #5851 #6035 and #6752.

## Chef Version
chef-client 13.8.11
Not present in 12.21.26

## Platform Version
Centos 7.4

## Replication Case
relevant client.rb settings
```
# client.rb

log_level :info
log_location STDOUT
```
run `/opt/chef/embedded/bin/chef-client -c /etc/chef/client.rb`
and compare to the output from `/opt/chef/embedded/bin/chef-client -c /etc/chef/client.rb | head -30`

## Client Output
Print statement debugging:
```
diff --git a/lib/chef/application.rb b/lib/chef/application.rb
index b20766c..b94f398 100644
--- a/lib/chef/application.rb
+++ b/lib/chef/application.rb
@@ -151,9 +151,16 @@ def set_specific_recipes
     def configure_logging
       configure_log_location
       Chef::Log.init(MonoLogger.new(Chef::Config[:log_location]))
+      puts "NAOMI: Chef::Config[:log_location].is_a?(IO) is #{Chef::Config[:log_location].is_a?(IO)}"
+      puts "NAOMI: Chef::Config[:log_location].tty? is #{Chef::Config[:log_location].tty?}"
+      puts "NAOMI: !( Chef::Config[:log_location].is_a?(IO) && Chef::Config[:log_location].tty? ) is #{!( Chef::Config[:log_location].is_a?(IO) && Chef::Config[:log_location].tty?
+      puts "NAOMI: !Chef::Config[:daemonize] is #{!Chef::Config[:daemonize]}"
+      puts "NAOMI: want_additional_logger is #{want_additional_logger?}"
+      puts "NAOMI: Chef::Log.loggers before configure stdout logger looks like #{Chef::Log.loggers}"
       if want_additional_logger?
         configure_stdout_logger
       end
+      puts "NAOMI: Chef::Log.loggers after configure stdout logger looks like #{Chef::Log.loggers}"
       Chef::Log.level = resolve_log_level
     rescue StandardError => error
       Chef::Log.fatal("Failed to open or create log file at #{Chef::Config[:log_location]}: #{error.class} (#{error.message})")
```
if I run `/opt/chef/embedded/bin/chef-client -c /etc/chef/client.rb` without redirecting STDOUT the logs look fine:
```
[2018-04-24T11:27:09-07:00] INFO: Loading Chef event handlers
NAOMI: Chef::Config[:log_location].is_a?(IO) is true
NAOMI: Chef::Config[:log_location].tty? is true
NAOMI: !( Chef::Config[:log_location].is_a?(IO) && Chef::Config[:log_location].tty? ) is false
NAOMI: !Chef::Config[:daemonize] is true
NAOMI: want_additional_logger is false
NAOMI: Chef::Log.loggers before configure stdout logger looks like [#<MonoLogger:0x0000000002b9a9e0 @progname=nil, @level=2, @default_formatter=#<Logger::Formatter:0x0000000002b9a9b8 @datetime_format=nil>, @formatter=#<Mixlib::Log::Formatter:0x0000000002b9a670 @datetime_format=nil>, @logdev=#<MonoLogger::LocklessLogDevice:0x0000000002b9a990 @shift_size=nil, @shift_age=nil, @filename=nil, @dev=#<IO:<STDOUT>>>>]
NAOMI: Chef::Log.loggers after configure stdout logger looks like [#<MonoLogger:0x0000000002b9a9e0 @progname=nil, @level=2, @default_formatter=#<Logger::Formatter:0x0000000002b9a9b8 @datetime_format=nil>, @formatter=#<Mixlib::Log::Formatter:0x0000000002b9a670 @datetime_format=nil>, @logdev=#<MonoLogger::LocklessLogDevice:0x0000000002b9a990 @shift_size=nil, @shift_age=nil, @filename=nil, @dev=#<IO:<STDOUT>>>>]
[2018-04-24T11:27:09-07:00] INFO: Forking chef instance to converge...
[2018-04-24T11:27:09-07:00] INFO: Chef event run_start fired
[2018-04-24T11:27:09-07:00] INFO: *** Chef 13.8.11 ***
[2018-04-24T11:27:09-07:00] INFO: Platform: x86_64-linux
[2018-04-24T11:27:09-07:00] INFO: Chef-client pid: 2190917
```
but if I redirect STDOUT by running `/opt/chef/embedded/bin/chef-client -c /etc/chef/client.rb | head -30 ` the logs duplicate:
```
[2018-04-24T11:31:59-07:00] INFO: Loading Chef event handlers
NAOMI: Chef::Config[:log_location].is_a?(IO) is true
NAOMI: Chef::Config[:log_location].tty? is false
NAOMI: !( Chef::Config[:log_location].is_a?(IO) && Chef::Config[:log_location].tty? ) is true
NAOMI: !Chef::Config[:daemonize] is true
NAOMI: want_additional_logger is true
NAOMI: Chef::Log.loggers before configure stdout logger looks like [#<MonoLogger:0x00000000029514e0 @progname=nil, @level=2, @default_formatter=#<Logger::Formatter:0x00000000029514b8 @datetime_format=nil>, @formatter=#<Mixlib::Log::Formatter:0x00000000029510a8 @datetime_format=nil>, @logdev=#<MonoLogger::LocklessLogDevice:0x0000000002951490 @shift_size=nil, @shift_age=nil, @filename=nil, @dev=#<IO:<STDOUT>>>>]
NAOMI: Chef::Log.loggers after configure stdout logger looks like [#<MonoLogger:0x00000000029514e0 @progname=nil, @level=2, @default_formatter=#<Logger::Formatter:0x00000000029514b8 @datetime_format=nil>, @formatter=#<Mixlib::Log::Formatter:0x00000000029510a8 @datetime_format=nil>, @logdev=#<MonoLogger::LocklessLogDevice:0x0000000002951490 @shift_size=nil, @shift_age=nil, @filename=nil, @dev=#<IO:<STDOUT>>>>, #<MonoLogger:0x0000000002950a90 @progname=nil, @level=0, @default_formatter=#<Logger::Formatter:0x0000000002950a68 @datetime_format=nil>, @formatter=#<Mixlib::Log::Formatter:0x00000000029510a8 @datetime_format=nil>, @logdev=#<MonoLogger::LocklessLogDevice:0x0000000002950a40 @shift_size=nil, @shift_age=nil, @filename=nil, @dev=#<IO:<STDOUT>>>>]
[2018-04-24T11:31:59-07:00] INFO: Forking chef instance to converge...
[2018-04-24T11:31:59-07:00] INFO: Forking chef instance to converge...
[2018-04-24T11:31:59-07:00] INFO: Chef event run_start fired
[2018-04-24T11:31:59-07:00] INFO: *** Chef 13.8.11 ***
[2018-04-24T11:31:59-07:00] INFO: *** Chef 13.8.11 ***
[2018-04-24T11:31:59-07:00] INFO: Platform: x86_64-linux
[2018-04-24T11:31:59-07:00] INFO: Platform: x86_64-linux
[2018-04-24T11:31:59-07:00] INFO: Chef-client pid: 2203775
[2018-04-24T11:31:59-07:00] INFO: Chef-client pid: 2203775
[2018-04-24T11:31:59-07:00] INFO: The plugin path /etc/chef/ohai/plugins does not exist. Skipping...
[2018-04-24T11:31:59-07:00] INFO: The plugin path /etc/chef/ohai/plugins does not exist. Skipping...
```

**Comment by lamont-granquist** [comment]
So I tried to fix the double-logging bug here:

https://github.com/chef/chef/pull/6752/files

The issue that I was running into was that for some odd reason `STDOUT` coming from `Chef::Config[:log_location]` as the default value was not matching STDOUT in the expression:  `Chef::Config[:log_location] != STDOUT`

If you look back in time I was thinking of removing this feature entirely, but then replaced it:

https://github.com/chef/chef/commit/4c01ebf6a35f221461c93bd915312f2703af8b29

You have to go back to the original commit by dan some 5-6 years ago for chef 11.0.0 for the full rationale of why the code is the way it is.

What I was concerned about by reverting the removal is people who setup syslog loggers in client.rb using `Chef::Config[:log_location]` and they will need a duplicate logger, or else they'll get syslog output and nothing on STDOUT when running on the command line.

I think the intent of this check:

```
( Chef::Config[:log_location] != STDOUT )
```

Can be more accurately described as "if the log_location is not set to the default value".  Now i think that Chef::Config supports checking if the value is the default or not via APIs in mixlib-config itself.  If i've recalled that correctly, those could be used to replace it with a simple `!Chef::Config.is_default_value(:log_location)` check.  But I haven't had the time to research if my recollection is correct or not (or if that'll actually work at all).

That would be my suggestion of the direction to start investigating though.

**Comment by tas50** [comment]
@NaomiReeves This should be fixed in the upcoming 14.5 release.
